staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -31,6 +31,77 @@ type DelayingInterface interface {
 	Interface
 	// AddAfter adds an item to the workqueue after the indicated duration has passed
 	AddAfter(item interface{}, duration time.Duration)
+	// AddWithOptions performs an add with additional implementation specific options/features.
+	AddWithOptions(item interface{}, opts AddOptions)
+	// AddAfterWithOptions performs a delayed add with additional implemententation specific options/features.
+	AddAfterWithOptions(item interface{}, duration time.Duration, opts AddAfterOptions)
```

Comment:
Possibly! :) but it will make the options themselves a lot more complicated because some options will apply to immediate behaviour and others to just the delayed behaviour.   I worry then about clients of the package mishandling the options or we would possibly need to start validating the options and returning an error?